src/features/runtime-checks.js

```diff
@@ -274,19 +285,12 @@ class DDGRuntimeChecks extends HTMLElement {
     }
 
     remove () {
-        const el = this._getElement()
-        if (el) {
-            return el.remove()
-        }
-        return super.remove()
+        return this._callMethod('remove')
     }
 
+    // @ts-expect-error TS node return here
     removeChild (child) {
-        const el = this._getElement()
-        if (el) {
-            return el.removeChild(child)
-        }
-        return super.removeChild(child)
+        return this._callMethod('removeChild', child)
```

Comment:
@shakyShane I couldn't get the type magic to align here.

The error is:
```
Property 'removeChild' in type 'DDGRuntimeChecks' is not assignable to the same property in base type 'HTMLElement'.
  Type '(child: any) => Node' is not assignable to type '<T extends Node>(child: T) => T'.
    Type 'Node' is not assignable to type 'T'.
      'Node' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'Node'.ts(2416)
```

This also breaks the `customElements.define` method below also.

Reply:
Yeah I tried assigning to many other things, I think the only way is perhaps adding a T on the class itself. Let me try that just in case it's that simple.

We have tests for many of these functions, I don't think this one specifically.

Reply:
Yeah didn't work, I'm just going to merge for now. The important methods like setAttribute have tests and correctly break if not working etc.